src/main/java/com/google/opensesame/projects/ProjectQuery.java

```diff
@@ -0,0 +1,118 @@
+package com.google.opensesame.projects;
+
+import static com.googlecode.objectify.ObjectifyService.ofy;
+
+import com.google.opensesame.util.ServletValidationException;
+import com.googlecode.objectify.annotation.Index;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Helper functions for querying the projects Datastore. */
+public class ProjectQuery {
+  public static final String FILTER_QUERY_PARAM = "filter";
```

Comment:
> However, in this case I don't think that this value will ever change since it is part of a well-established API.

Just to clarify, this is being used by my function `ProjectQuery.getQueryFiltersFromRequest`. Would you mind elaborating on what you mean by "part of a well-established API"?

Also, my reasoning for extracting it into a constant was so that future developers wouldn't have to look deep into the code to determine what the URL parameter is for sending filters. Is there maybe another way that I can make this value easy for developers to find?

Reply:
I think that it would be fairly easy to just search the file for "getParameterValues" using ctrl + f.

Reply:
Yeah that makes sense. I'll put it into the method, thanks!